### PR TITLE
Implement places list screen.

### PR DESCRIPTION
**Pull request by kaushiksaliya**
Implement places list screen.

Screenshot from 2024-06-21 16-58-15Screenshot from 2024-06-20 16-56-28

### CODE REVIEW

File: app/assets/locales/app_en.arb [hunk 1]
```diff
@@ -141,5 +141,18 @@
...
   "contact_support_attachment": "Attachment (if any)",
   "contact_support_submit_title": "Submit",
-  "contact_support_feedback_alert_message": "Thanks! Your feedback has been recorded."
+  "contact_support_feedback_alert_message": "Thanks! Your feedback has been recorded.",
+
+  "@_PLACES": {
+  },
+  "places_list_title": "Places",
+  "places_list_add_place_btn_text": "Add place",
+  "places_list_suggestion_add_your_place":"Add your {place}",
```
Comment (cp-radhika-s) [inline app/assets/locales/app_en.arb:150]: Let's remove `your`

File: app/lib/ui/flow/geofence/places/places_list_view.dart [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            if (state.currentUser?.id == place.created_by &&
+                suggestionName == placeName) {
+              return false;
+            }
+          }
+          return true;
+        }).toList();
+
+        setState(() {
+          suggestions = newSuggestedPlace;
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/geofence/places/places_list_view.dart:223]: We should move suggestion state to view notifier, notifier will decide suggestion not the UI

File: app/lib/ui/flow/geofence/places/places_list_view_model.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        error: error,
+        stackTrace: stack,
+      );
+    }
+  }
+
+  void onClickDeletePlace(ApiPlace place) {
+    state = state.copyWith(
+      placesToDelete: place,
+      showDeletePlaceDialog: DateTime.now(),
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/geofence/places/places_list_view_model.dart:46]: Shouldn't it be bool?

File: data/lib/service/place_service.dart [hunk 4]
```diff
@@ -25,4 +25,8 @@ class PlaceService {
         .map((doc) => ApiPlace.fromJson(doc.data() as Map<String, dynamic>))
         .toList());
   }
+
+  Future<void> deletePlace(String spaceId, String placeId) async {
```
Comment (cp-radhika-s) [inline data/lib/service/place_service.dart:29]: Also need to remove it's sub collections

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

File: app/lib/ui/flow/geofence/places/places_list_view_model.dart (hunk 3 above)
Comment (kaushiksaliya) [reply app/lib/ui/flow/geofence/places/places_list_view_model.dart:46]: if we make it bool then we need to call function to false state after dialog close but here not.

**Review by cp-radhika-s — APPROVED** [review]

**Pull request merged**